src/Cli.Tests/EnvironmentTests.cs

```diff
@@ -162,8 +164,7 @@ public async Task FailureToStartEngineWhenEnvVarNamedWrong()
         );
 
         string? output = await process.StandardError.ReadLineAsync();
-        StringAssert.Contains(output, "Environmental Variable, "
-            + expectedEnvVarName + ", not found.", StringComparison.Ordinal);
+        Assert.AreEqual(output, "Deserialization of the configuration file failed.");
```

Comment:
The previous check explicitly checked for the failure that the test was trying to trigger. Can you include both your new addition to check the high level deserialization failure notice as well as checking `StringAssert.Contains(output, "Environmental Variable, "`?

Also, the test name indicates that this is testing an invalid env_var name provided in config, as opposed to a invalid connection string failing post-processing. Can you clarify test intent here?

Reply:
The error message contains whatever we decide to put in the messaging, which right now is, "Deserialization of the configuration file failed.", to add more details than that we would have to add it to the original message.

Reply:
Understood that we have error handling now and set the message "Deserialization of the configuration file failed." My point is that the test can also check the specific error causing the deserialization issue. Those details are printed to the console as shown by your screenshot: 
![image](https://github.com/Azure/data-api-builder/assets/6414189/403bde82-3fb6-4abb-9926-92293ed54334)

The environment variable error messaging may still be available to check here, perhaps if you continue reading errors printed to the console `process.StandardError.ReadLineAsync();`.

That way, the test can be specific so you know that no other possible issues are causing the test to pass/fail:
- if you're testing env_var issues, check for both the "Deserialization of the configuration file failed." and exception specific messaging printed to the console
- if you're testing connectionstring builder errors, check for both the "Deserialization of the configuration file failed." and exception specific messaging printed to the console

Reply:
sure we can read out the entire error message and verify the underlying exception includes the environment variable error.